### PR TITLE
[Agent] Fix installers

### PR DESCRIPTION
**Pull request by michalpristas**
## What does this PR do?

This PR changes installers  in two ways
#### SNAPSHOT handling
for now  turned on by default, if we want to enable it only for snapshot agent we need to modify build to set flag at build time  and then enable snapshot installer afterwards

#### windows refactor
install step on windows was not working properly, while we were unzipping we were creating one additional directory layer, this is now removed.

#### Small config change
Default configs are not using 9300 endpoint for ES from now on

## Why is it important?

To simplify use case  when running from snapshot and entire windows flow

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have made corresponding change to the default configuration files
- [ ] I have added tests that prove my fix is effective or that my feature works

#### Scenario 1

run `SNAPSHOT=true mage package` to build snapshot package
unpack it somewhere locally and try if it runs with no changes to config

#### Scenario 2
Same as 1 but on windows, windows was broken even for non SNAPSHOT packages.

### CODE REVIEW

File: x-pack/agent/pkg/artifact/install/zip/zip_installer.go [hunk 1]
```diff
@@ -58,14 +58,25 @@ func (i *Installer) unzip(programName, version, installPath string) error {
...
-	powershellCmd := fmt.Sprintf(powershellCmdTemplate, installPath, programName)
+func (i *Installer) runInstall(programName, version, installPath string) error {
+	alignedPath := alignInstallPath(installPath, version)
+	powershellCmd := fmt.Sprintf(powershellCmdTemplate, alignedPath, programName)
 
 	installCmd := exec.Command("powershell", "-command", powershellCmd)
 	return installCmd.Run()
 }
+
+func alignInstallPath(installPath, version string) string {
```
Comment (michalpristas) [inline x-pack/agent/pkg/artifact/install/zip/zip_installer.go]: this i dont like i will rewrite this part

**Comment by michalpristas** [comment]
Jenkins test this

**Comment by ph** [comment]
failures doesn't seem related to this PR?

**Comment by ph** [comment]
backported in https://github.com/elastic/beats/pull/17574 to have a working build

**Pull request merged**